stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -702,9 +702,10 @@ end
 function opnorm2(A::AbstractMatrix{T}) where T
     require_one_based_indexing(A)
     m,n = size(A)
-    if m == 1 || n == 1 return norm2(A) end
     Tnorm = typeof(float(real(zero(T))))
-    (m == 0 || n == 0) ? zero(Tnorm) : convert(Tnorm, svdvals(A)[1])
+    if m == 0 || n == 0 return zero(Tnorm) end
+    if m == 1 || n == 1 return norm2(A) end
```

Comment:
Do we want a `convert(Tnorm, norm2(A))` just to be paranoid about type-stability?  (I'm not sure why it's required for the `svdvals` either.)

Reply:
I've removed the convert on the svdvals and added inference tests.